contracts/facets/EmbalmerFacet.sol

```diff
@@ -0,0 +1,130 @@
+    /// @param sarcoToken The erc20 sarcophagus token
+    /// @param canBeTransfered Whether the sarcophagus can be transferred
+    /// @return The index of the new sarcophagus
+    function initializeSarcophagus(
+        string memory name,
+        LibTypes.Archaeologist[] memory archaeologists,
+        address arweaveArchaeologist,
+        address recipient,
+        uint256 resurrectionTime,
+        bytes32 identifier,
+        IERC20 sarcoToken,
+        bool canBeTransfered
+    ) external returns (uint256) {
+        // Confirm that this exact sarcophagus does not already exist
+        require(
+            !s.sarcophaguses[identifier].exists,
+            "sarcophagus already exists"
+        );
+
+        // Confirm that the ressurection time is in the future
+        require(
+            resurrectionTime > block.timestamp,
+            "resurrection time must be in the future"
+        );
+
+        // Initialize a list of archaeologist addresses
+        address[] memory archaeologistAddresses = new address[](
+            archaeologists.length
+        );
+
+        for (uint256 i = 0; i < archaeologists.length; i++) {
+            // Calculate the amount of cursed bond the archaeologists needs to lock up
+            uint256 cursedBondAmount = LibBonds.calculateCursedBond(
+                archaeologists[i].diggingFee,
+                archaeologists[i].bounty
+            );
+
+            // Confirm that the archaeologist has enough free bond.
+            // This error could mean that the archaeologist has either run out
+            // of free bond or has never even interacted with sarcophagus
+            require(
+                s.freeBonds[archaeologists[i].archAddress] >= cursedBondAmount,
+                "archaeologist does not have enough free bond"
+            );
+
+            // Lock up the archaeologist's bond by the cursed bond amount
+            LibBonds.lockUpBond(
+                archaeologists[i].archAddress,
+                cursedBondAmount
+            );
+
+            // Add to the neccessary data structures
+            s.archaeologistSarcophaguses[archaeologists[i].archAddress].push(
+                identifier
+            );
+            s.sarcophagusArchaeologists[identifier][
+                archaeologists[i].archAddress
+            ] = archaeologists[i];
+
+            // Add the archaeologist address to the list of addresses to be
+            // passed in to the sarcophagus object
+            archaeologistAddresses[i] = archaeologists[i].archAddress;
+        }
+
+        s.sarcophaguses[identifier] = LibTypes.Sarcophagus({
+            name: name,
+            exists: true,
+            canBeTransferred: canBeTransfered,
+            resurrectionTime: resurrectionTime,
+            arweaveTxId: "",
+            embalmer: msg.sender,
+            recipientAddress: recipient,
+            arweaveArchaeologist: arweaveArchaeologist,
+            archaeologists: archaeologistAddresses
```

Comment:
Do we need to store these addresses here, since they are stored in the `sarcophagusArchaeologists` mapping? Or could we store the archaeologist type array here instead of the addresses (since the type includes the address) and remove the mapping?

Reply:
Addressing the first part, if we don't store an array of archaeologists (or an array of structs) then how would we know which archaeologists are on the sarcophagus? Since the `sarcophagusArchaeologists` is a mapping there is no way to iterate over it. The best we could do is, given an archaeologist address, tell if the archaeologist is on that sarcophagus.

The same applies for the other addresses: embalmer, recipient, and arweaveArchaeologist. If these are not stored on the Sarcophagus the best we could do is, given an address, tell if that address belongs to the sarcophagus. 

Addressing the second part, storing an array of Archaeologist structs in the Sarcophagus object is possible. However, the pattern I'm seeing in solidity is to not store an array of structs but an array of keys (in this case the archaeologist addresses) and then store a mapping of the structs somewhere else.

Please correct me if I'm wrong on any of this.

Reply:
That all makes sense -- if storing an array of structs is an anti-pattern, then makes sense to not do it here. This can stay as written, thanks for laying that out.